src/app/application.cpp

```diff
@@ -307,21 +307,21 @@ void Application::runExternalProgram(BitTorrent::TorrentHandle *const torrent) c
 #endif
 }
 
-void Application::sendNotificationEmail(BitTorrent::TorrentHandle *const torrent)
+void Application::sendNotificationEmail(const BitTorrent::TorrentHandle *torrent) const
 {
     // Prepare mail content
-    QString content = QObject::tr("Torrent name: %1").arg(torrent->name()) + "\n";
-    content += QObject::tr("Torrent size: %1").arg(Utils::Misc::friendlyUnit(torrent->wantedSize())) + "\n";
-    content += QObject::tr("Save path: %1").arg(torrent->savePath()) + "\n\n";
-    content += QObject::tr("The torrent was downloaded in %1.",
-                  "The torrent was downloaded in 1 hour and 20 seconds")
-            .arg(Utils::Misc::userFriendlyDuration(torrent->activeTime())) + "\n\n\n";
-    content += QObject::tr("Thank you for using qBittorrent.") + "\n";
+    const QString content = QObject::tr("Torrent name: %1").arg(torrent->name()) + "\n"
+        + QObject::tr("Torrent size: %1").arg(Utils::Misc::friendlyUnit(torrent->wantedSize())) + "\n"
+        + QObject::tr("Save path: %1").arg(torrent->savePath()) + "\n\n"
+        + QObject::tr("The torrent was downloaded in %1.", "The torrent was downloaded in 1 hour and 20 seconds")
+            .arg(Utils::Misc::userFriendlyDuration(torrent->activeTime())) + "\n\n\n"
+        + QObject::tr("Thank you for using qBittorrent.") + "\n";
 
     // Send the notification email
-    Net::Smtp *sender = new Net::Smtp;
-    sender->sendMail("notification@qbittorrent.org",
-                     Preferences::instance()->getMailNotificationEmail(),
+    const Preferences *pref = Preferences::instance();
+    Net::Smtp *smtp = new Net::Smtp;
```

Comment:
memory leak

Reply:
What about: https://github.com/qbittorrent/qBittorrent/blob/master/src/base/net/smtp.cpp#L115

Reply:
or is it a good idea to pass parent pointer to `Smtp` regardless?

Reply:
Valgrind shows memleak here. Maybe its algorithm makes a mistake?

Reply:
BTW cppcheck reports it too...